plugins/block-dynamic-connection/src/dynamic_if.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Defines a version of the if block with dyanmic
+ *     inputs that appear when a block is dragged over inputs on the block.
+ */
+
+import * as Blockly from 'blockly/core';
+
+Blockly.Blocks['dynamic_if'] = {
+  /**
+   * Counter for the next input to add to this block.
+   * @type {number}
+   */
+  inputCounter: 1,
+
+  /**
+   * Minimum number of inputs for this block.
+   * @type {number}
+   */
+  minInputs: 1,
+  /**
+   * Block for concatenating any number of strings.
+   * @this {Blockly.Block}
+   */
+  init: function() {
+    this.setHelpUrl(Blockly.Msg['CONTROLS_IF_HELPURL']);
+    this.setStyle('logic_blocks');
+
+    this.appendValueInput('IF0')
+        .setCheck('Boolean')
+        .appendField(Blockly.Msg['CONTROLS_IF_MSG_IF'], 'if');
+    this.appendStatementInput('DO0');
```

Comment:
We need to add the text field for the "do" that sits in front of the statement inputs. The message string is `CONTROLS_IF_MSG_THEN`.

We also need to add the next and previous statement to the block.
```
this.setNextStatement(true);
this.setPreviousStatement(true);
```